src/daemon/default_vm_image_vault.cpp

```diff
@@ -522,10 +522,18 @@ void mp::DefaultVMImageVault::update_images(const FetchType& fetch_type, const P
         if (record.second.query.query_type == Query::Type::Alias &&
             record.first.compare(0, record.second.query.release.length(), record.second.query.release) != 0)
         {
-            auto info = info_for(record.second.query);
-            if (info.id.toStdString() != record.first)
+            try
             {
-                keys_to_update.push_back(record.first);
+                auto info = info_for(record.second.query);
+                if (info.id.toStdString() != record.first)
+                {
+                    keys_to_update.push_back(record.first);
+                }
+            }
+            catch (const std::exception& e)
```

Comment:
@ricab, I understand what you are saying, but the point of this change is to allow `update_images()` to continue regardless of the exception thrown which is what this is doing here.

Reply:
I'm with @ricab here, we should only ignore known exceptions. Ignoring wholesale may very well lead to bugs much more difficult to track.

Reply:
Why would you want to continue regardless of the exception? I think you are overlooking that there could be exceptions meaning a bad program state or something, on which we should fail early. 

[Here](https://stackoverflow.com/questions/21938/is-it-really-that-bad-to-catch-a-general-exception) is one of the first google results discussing this. There are multiple good opinions there. I'll just quote from the first answer:

_A good rule of thumb is that you should only catch exceptions that you can properly deal with yourself._

Reply:
Ok, I'm overruled.  Will change this.

Reply:
I will say though, if this is a problem for this PR, then we really need to audit the code since I believe this type of general catching is all over the place.